app/lib/submission_builder/ty2022/states/authentication_header.rb

```diff
@@ -53,25 +53,28 @@ def document
             end
             xml.TransmissionDetail do
               xml.InitialCreation do
+                device_info = @submission.data_source.initial_efile_device_info
+                # ip_for_irs for test env
                 xml.IPAddress do
-                  xml.IPv4AddressTxt '1.2.3.4'
+                  xml.IPv4AddressTxt device_info&.ip_address if device_info.ip_address.ipv4?
+                  xml.IPv6AddressTxt device_info&.ip_address if device_info.ip_address.ipv6?
                 end
-                xml.IPTs '2023-10-10T12:00:00-05:00'
-                xml.DeviceId 'AB' * 20
-                xml.DeviceTypeCd 'Desktop'
-                xml.IPPortNum '1234'
+                xml.IPTs datetime_type(device_info&.created_at)
+                xml.DeviceId device_info&.device_id || 'AB' * 20
```

Comment:
noticed we did this `|| 'AB' * 20` in CTC land presumably to allow clients to submit a return even without a device id